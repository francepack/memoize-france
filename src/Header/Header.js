import React, { Component } from 'react';
import Directions from  '../Directions/Directions';

export default class Header extends Component {
  constructor() {
    super();
    this.state = {
      displayDirections: false
    }
  }

  toggleDirections = () => {
    this.setState({ displayDirections: !this.state.displayDirections })
  }

  render() {
    return(
      <header className="header">
        <h1 className="title">codeQuiz</h1>
        <button className="direction-btn" onClick={this.toggleDirections}>Info</button>
        <p className="subtitle">Find a subject to study and test your knowledge</p>
        <Directions 
          displayDirections={this.state.displayDirections}
          toggleDirections={this.toggleDirections}
        />
      </header>
    )
  }
}
        