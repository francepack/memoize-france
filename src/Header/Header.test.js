import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const toggleDirections = jest.fn();

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Header />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default states', () => {
    expect(wrapper.state()).toEqual({
      displayDirections: false
    });
  });

  it('should update displayDirections state when toggleDirections is invoked', () => {
    expect(wrapper.state('displayDirections')).toEqual(false);
    wrapper.instance().toggleDirections();
    expect(wrapper.state('displayDirections')).toEqual(true);
  });

  it('should invoke toggleDirections when the directions button is clicked', () => {
    wrapper.find('.direction-btn').simulate('click', { preventDefault: () => {}});
    expect(toggleDirections).toBeCalled(); 
  });

});  