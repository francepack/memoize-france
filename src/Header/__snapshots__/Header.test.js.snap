// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
        >
          codeQuiz
        </h1>,
        <button
          className="direction-btn"
          onClick={[Function]}
        >
          Info
        </button>,
        <p
          className="subtitle"
        >
          Find a subject to study and test your knowledge
        </p>,
        <Directions
          displayDirections={false}
          toggleDirections={[Function]}
        />,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "codeQuiz",
          "className": "title",
        },
        "ref": null,
        "rendered": "codeQuiz",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Info",
          "className": "direction-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Info",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find a subject to study and test your knowledge",
          "className": "subtitle",
        },
        "ref": null,
        "rendered": "Find a subject to study and test your knowledge",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "displayDirections": false,
          "toggleDirections": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
          >
            codeQuiz
          </h1>,
          <button
            className="direction-btn"
            onClick={[Function]}
          >
            Info
          </button>,
          <p
            className="subtitle"
          >
            Find a subject to study and test your knowledge
          </p>,
          <Directions
            displayDirections={false}
            toggleDirections={[Function]}
          />,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "codeQuiz",
            "className": "title",
          },
          "ref": null,
          "rendered": "codeQuiz",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Info",
            "className": "direction-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Info",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find a subject to study and test your knowledge",
            "className": "subtitle",
          },
          "ref": null,
          "rendered": "Find a subject to study and test your knowledge",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "displayDirections": false,
            "toggleDirections": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
